src/dev/code_coverage/ingest_coverage/transforms.js

```diff
@@ -0,0 +1,160 @@
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { left, right } from './either';
+import { always, id, noop } from './utils';
+
+const maybeTotal = x => (x === 'total' ? left(x) : right(x));
+
+export const trimLeftFrom = (text, x) => x.replace(new RegExp(`(?:.*)(${text}.*$)`, 'gm'), '$1');
+
+export const statsAndstaticSiteUrl = (...xs) => {
+  const [staticSiteUrl] = xs[0][1];
+  const [stats] = xs[0];
+  return {
+    staticSiteUrl,
+    ...stats,
+  };
+};
+
+export const addJsonSummaryPath = jsonSummaryPath => obj => ({
+  jsonSummaryPath: trimLeftFrom('target', jsonSummaryPath),
+  ...obj,
+});
+
+export const truncate = text => obj => {
+  const { staticSiteUrl } = obj;
+  if (staticSiteUrl.includes(text)) obj.staticSiteUrl = trimLeftFrom(text, staticSiteUrl);
+  return obj;
+};
+
+export const addTimeStamp = ts => obj => ({
+  ...obj,
+  '@timestamp': ts,
+});
+
+const captureAfterRootFolder = /.*kibana(.*$)/;
+const afterRootFolder = x => captureAfterRootFolder.exec(x)[1];
+const fixFront = x => afterRootFolder(x);
+const setTotal = x => obj => (obj.isTotal = x);
+const mutateTrue = setTotal(true);
+const mutateFalse = setTotal(false);
+
+const root = urlBase => ts => testRunnerType =>
+  `${urlBase}/${ts}/${testRunnerType.toLowerCase()}-combined`;
+
+const prokForTotalsIndex = mutateTrue => urlRoot => obj =>
+  right(obj)
+    .map(mutateTrue)
+    .map(always(`${urlRoot}/index.html`))
+    .fold(noop, id);
+
+const prokForCoverageIndex = mutateFalse => urlRoot => obj => siteUrl =>
+  right(siteUrl)
+    .map(x => {
+      mutateFalse(obj);
+      return x;
+    })
+    .map(fixFront)
+    .map(x => `${urlRoot}${x}.html`)
+    .fold(noop, id);
+
+export const staticSite = urlBase => obj => {
+  const { staticSiteUrl, testRunnerType } = obj;
+  const ts = obj['@timestamp'];
+  const urlRoot = root(urlBase)(ts)(testRunnerType);
+  const prokTotal = prokForTotalsIndex(mutateTrue)(urlRoot);
+  const prokCoverage = prokForCoverageIndex(mutateFalse)(urlRoot)(obj);
+  const prokForBoth = always(maybeTotal(staticSiteUrl).fold(always(prokTotal(obj)), prokCoverage));
+
+  return { ...obj, staticSiteUrl: prokForBoth() };
+};
+
+export const coveredFilePath = obj => {
+  const { staticSiteUrl } = obj;
+
+  const withoutCoveredFilePath = always(obj);
+  const dropFront = x => trimLeftFrom('/kibana/', x).replace(/(^\/kibana\/)/, '');
```

Comment:
@wayneseymour I think we cannot rely on `kibana` text in path because of multiple occurrences, but to be 100% safe we can get and store prefix path via `${pwd}` that was used on merge phase (`/var/lib/jenkins/workspace/elastic+kibana+code-coverage/` ) and pass it as argument to ingest script?